server/service/src/sync/translation_and_integration.rs

```diff
@@ -107,38 +95,66 @@ impl<'a> TranslationAndIntegration<'a> {
         };
 
         for (number_of_records_integrated, sync_record) in sync_records.into_iter().enumerate() {
-            let translation_result = match self.translate_sync_record(&sync_record, translators) {
-                Ok(translation_result) => translation_result,
-                // Record error in sync buffer and in result, continue to next sync_record
-                Err(translation_error) => {
-                    self.sync_buffer
-                        .record_integration_error(&sync_record, &translation_error)?;
-                    result.insert_error(&sync_record.table_name);
-                    warn!(
-                        "{:?} {:?} {:?}",
-                        translation_error, sync_record.record_id, sync_record.table_name
-                    );
-                    // Next sync_record
-                    continue;
+            let pull_translation_results =
+                match self.translate_sync_record(&sync_record, translators) {
+                    Ok(translation_result) => translation_result,
+                    // Record error in sync buffer and in result, continue to next sync_record
+                    Err(translation_error) => {
+                        self.sync_buffer
+                            .record_integration_error(&sync_record, &translation_error)?;
+                        result.insert_error(&sync_record.table_name);
+                        warn!(
+                            "{:?} {:?} {:?}",
+                            translation_error, sync_record.record_id, sync_record.table_name
+                        );
+                        // Next sync_record
+                        continue;
+                    }
+                };
+
+            let mut integration_records = Vec::new();
+            let mut ignored = false;
+            for pull_translation_result in pull_translation_results {
+                match pull_translation_result {
+                    PullTranslateResult::IntegrationOperations(mut operations) => {
+                        integration_records.append(&mut operations)
+                    }
+                    PullTranslateResult::Ignored(ignore_message) => {
+                        ignored = true;
+                        // note, this might get cleared if other translator produced some ops
+                        self.sync_buffer.record_integration_error(
+                            &sync_record,
+                            &anyhow::anyhow!("Ignored: {}", ignore_message),
+                        )?;
+                        result.insert_error(&sync_record.table_name);
+
+                        debug!(
+                            "Ignored record: {:?} {:?} {:?}",
+                            ignore_message, sync_record.record_id, sync_record.table_name
+                        );
+                        continue;
+                    }
+                    PullTranslateResult::NotMatched => {}
```

Comment:
I feel like we want a message for `NotMatched` as well... currently it just says `Translator for record not found`.